Add created at to all products created

jamescowie opened the pull request:
Resolves the null date issue reported within magento 1.9 on product creation using the product fixture.

jamescowie committed:
Add created at to all products created

alistairstead commented on src/MageTest/MagentoExtension/Fixture/Product.php:
```diff
@@ -84,6 +84,7 @@ public function create(array $attributes)
                 return $website->getId();
             }, $websiteHelper->getWebsites()))
             ->setData($this->mergeAttributes($attributes))
+            ->setCreatedAt(null)
```
This will cause prophecy exceptions in the specs... They will need to be updated accordingly as part of the PR

jamescowie committed:
Update specs to enable mocking

jamescowie commented:
Hi @alistairstead I have updated the specs so that this should pass CI now. However if you can review again for me and provide feedback on the Mock that I added it would be very much appreciated.

alistairstead commented:
![tumblr_mszm52mzg21stn6klo1_400](https://cloud.githubusercontent.com/assets/62936/3454473/f9773a1e-01d1-11e4-8e02-f76f094cb47f.gif)
